### PR TITLE
Test errors for out-of-range limit requests

### PR DESCRIPTION
**Pull request by kainino0x**
See https://crbug.com/1499030

Issue: None

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/util/math.ts [hunk 1]
```diff
@@ -2056,6 +2056,7 @@ export function isPowerOfTwo(n: number): boolean {
   if (!Number.isInteger(n)) {
     return false;
   }
+  assert((n | 0) === n, 'isPowerOfTwo only supports 32-bit numbers');
```
Comment (kainino0x) [inline src/webgpu/util/math.ts:2059]: I verified that the tests that use this helper still pass.
Could switch to `Math.log2(n) % 1 === 0` but I'm not 100% if that's a sound implementation either so I just did this.
`webgpu:api,validation,capability_checks,limits,*`
`webgpu:api,operation,adapter,requestDevice:*`

File: src/webgpu/api/operation/adapter/requestDevice.spec.ts [hunk 2]
```diff
@@ -311,6 +311,61 @@ g.test('limit,better_than_supported')
...
+    const requiredLimits = {
+      [limit]: value,
+    };
+
+    const errorName =
+      value < 0 || value > Number.MAX_SAFE_INTEGER
+        ? 'TypeError'
+        : limitInfo.class === 'maximum' && value > adapter.limits[limit]
+        ? 'OperationError'
+        : limitInfo.class === 'alignment' && (value > 2 ** 31 || !isPowerOfTwo(value))
```
Comment (kainino0x) [inline src/webgpu/api/operation/adapter/requestDevice.spec.ts:362]: The spec does not actually currently say that `2**32` or `2**53` is an error for alignment limits. I'm proposing it to the spec:
https://github.com/gpuweb/gpuweb/pull/4358

**Review by greggman — COMMENTED** [review]
I am confused. The limits tests already test trying to go over the max limit and they all pass. 

https://github.com/gpuweb/cts/blob/41f89e77b67e6b66cb017be4e00235a0a9429ca7/src/webgpu/api/validation/capability_checks/limits/limit_utils.ts#L376

What is the bug this is trying to surface that wasn't already tested?

Can you just add a case there instead of making a new test?

**Comment by kainino0x** [comment]
> What is the bug this is trying to surface that wasn't already tested?

We mishandle values &ge; 2^32 for 32-bit limit types, by `static_cast<uint32_t>(uint64_t)`:
https://chromium-review.googlesource.com/c/chromium/src/+/5002999

> Can you just add a case there instead of making a new test?

I may have missed the appropriate test to modify, but I only saw tests that used `mul`/`add` and I couldn't easily get specific corner cases in those tests.
I could certainly reduce the number of subcases in this test though.

**Review by greggman — COMMENTED** [review]
ok, lgtm 🤷‍♂️

If you wanted to give it a try you could add a new limit here

https://github.com/gpuweb/cts/blob/41f89e77b67e6b66cb017be4e00235a0a9429ca7/src/webgpu/api/validation/capability_checks/limits/limit_utils.ts#L207

atLimit, overLimit, overStorageSize

but I this fine too I guess. It tests more values.

**Review by greggman — APPROVED** [review]

**Comment by kainino0x** [comment]
Ack, I think I'll keep this since it can easily test a lot of values. It also does fit in with the tests next to it (even if we have some redundancy between these and the limit_utils based tests).
